torch/_refs/__init__.py

```diff
@@ -2392,6 +2399,67 @@ def permute(a: TensorLikeType, dims: DimsSequenceType) -> TensorLikeType:
     return prims.transpose(a, _permutation)
 
 
+@register_decomposition(torch.ops.aten.repeat)
+def repeat(a: Tensor, *repeat_shape) -> Tensor:
+    # NOTE: cannot use utils.extract_shape_from_varargs here
+    # because that also validates the shape, but the shape
+    # given to repeat may be "invalid"
+    if len(repeat_shape) == 1 and isinstance(repeat_shape[0], Sequence):
+        repeat_shape = tuple(repeat_shape[0])
+
+    utils.check(
+        len(repeat_shape) >= len(a.shape),
+        lambda: "Number of dimensions of repeat dims can not be smaller than number of dimensions of tensor",
+    )
+
+    if len(repeat_shape) == 0:
+        return torch.clone(a)
+
+    num_new_dimensions = len(repeat_shape) - a.ndim
+    padded_shape = [1] * num_new_dimensions
+    for dim_size in a.shape:
+        padded_shape.append(dim_size)
+
+    target_shape = tuple(
+        padded_size * repeat_size
+        for padded_size, repeat_size in zip(padded_shape, repeat_shape)
+    )
+
+    # return an empty tensor if one of the repeat_shape dimensions is zero
+    if 0 in repeat_shape:
+        return torch.empty(
+            target_shape,
+            dtype=a.dtype,
+            device=a.device,
+            requires_grad=a.requires_grad,
```

Comment:
Not sure whether we should specify the memory format here? Probably unnecessary as there're 0s in `target_shape`
```suggestion
            requires_grad=a.requires_grad,
            memory_format=utils.suggest_memory_format(a),
```